### ISSUE TITLE
[BUG] Statistics outputs for fix shake and shake/kokkos are different

### ISSUE THREAD

**Issue by yueyericardo** [lammps/lammps#3730]
**Summary**

The Statistics outputs for fix shake and shake/kokkos are different

Without kokkos it prints the following, which has the correct number of corresponding bonds.
```
SHAKE stats (type/ave/delta/count) on step 0
Bond:   1   0.957199  0.00219791      1260
Bond:   2   1.0901    0.000781659       10
Bond:   3   1.00984   0.000681794        2
Per MPI rank memory allocation (min/avg/max) = 7.417 | 7.417 | 7.417 Mbytes
   Step         PotEng         KinEng         TotEng          Temp          Press          Volume
         0  -30516812       1708.8984     -30515103       385.54136     -11344.435      31392.599
       500  -30517612       1386.0261     -30516226       312.69874     -9370.9687      31392.599
SHAKE stats (type/ave/delta/count) on step 1000
Bond:   1   0.960001  7.00129e-06     1260
Bond:   2   1.09      9.40483e-07       10
Bond:   3   1.01      6.66134e-16        2
      1000  -30517756       1308.8299     -30516447       295.28264     -10290.978      31392.599
```

With kokkos, it only has bond type 2, and the count is wrong.
```
WARNING: Fixes cannot yet send exchange data in Kokkos communication, switching to classic exchange/border communication (src/KOKKOS/comm_kokkos.cpp:651)
SHAKE/KK stats (type/ave/delta/count) on step 0
Bond:   2   1.09021   0.000317385     1908
Per MPI rank memory allocation (min/avg/max) = 3.409 | 3.409 | 3.409 Mbytes
   Step         PotEng         KinEng         TotEng          Temp          Press          Volume
         0  -30516812       1708.8984     -30515103       385.54136     -11312.545      31392.599
       500  -30517640       1348.4325     -30516292       304.21733     -10074.678      31392.599
SHAKE/KK stats (type/ave/delta/count) on step 1000
Bond:   2   1.09      1.68146e-07     1908
      1000  -30517720       1318.6074     -30516401       297.48854     -9884.7019      31392.599
```

**LAMMPS Version and Platform**

```
Large-scale Atomic/Molecular Massively Parallel Simulator - 3 Aug 2022
Git info (HEAD / patch_3Aug2022-282-g56de9c6d66)
```

**Comment by stanmoore1** [comment]
I believe I fixed this already in #3719, can you try with latest `develop` branch and see if the issue persists? Thanks

**Comment by yueyericardo** [comment]
Thanks for your quick reply!
Confirm that this issue is gone in the latest `develop` branch.
